### PR TITLE
refactor: use new API to refactor existing components

### LINKED ISSUE #515: [Refactor] Data migration
### Description

As the feature implemented in https://github.com/X-lab2017/open-digger/pull/1036 , the data for OpenDigger will be in a new place with new identical form.

The basic URL will be `https://oss.x-lab.info/open_digger/github`, and there will be 16 repo metrics and 2 user metrics right now. The repo metric will be at `{basicUrl}/{owner}/{repo}/{metric}.json` and user metric will be at `{basicUrl}/{login}/{metric}.json`.

The repo metrics are:
- activity: https://oss.x-lab.info/open_digger/github/apache/flink/activity.json
- openrank: https://oss.x-lab.info/open_digger/github/apache/flink/openrank.json
- attention: https://oss.x-lab.info/open_digger/github/apache/flink/attention.json
- technical_fork: https://oss.x-lab.info/open_digger/github/apache/flink/technical_fork.json
- stars: https://oss.x-lab.info/open_digger/github/apache/flink/stars.json
- issues_new: https://oss.x-lab.info/open_digger/github/apache/flink/issues_new.json
- issues_closed: https://oss.x-lab.info/open_digger/github/apache/flink/issues_closed.json
- code_change_lines_add: https://oss.x-lab.info/open_digger/github/apache/flink/code_change_lines_add.json
- code_change_lines_remove: https://oss.x-lab.info/open_digger/github/apache/flink/code_change_lines_remove.json
- code_change_lines_sum: https://oss.x-lab.info/open_digger/github/apache/flink/code_change_lines_sum.json
- change_requests(open pr): https://oss.x-lab.info/open_digger/github/apache/flink/change_requests.json
- change_requests_accepted(merged pr): https://oss.x-lab.info/open_digger/github/apache/flink/change_requests_accepted.json
- issue_comments: https://oss.x-lab.info/open_digger/github/apache/flink/issue_comments.json
- change_requests_reviews(review comment): https://oss.x-lab.info/open_digger/github/apache/flink/change_requests_reviews.json
- participants: https://oss.x-lab.info/open_digger/github/apache/flink/participants.json
- bus_factor: https://oss.x-lab.info/open_digger/github/apache/flink/bus_factor.json

The user metrics are:
- activity: https://oss.x-lab.info/open_digger/github/frank-zsy/activity.json
- openrank: https://oss.x-lab.info/open_digger/github/frank-zsy/openrank.json

The upload will be completed in this Sunday but repo examples are valid now. And all the data is up to date.

I think we can start the data migration now and the data will be sufficient for Hypercrx now except for network data.

### PR DESCRIPTION
**Pull request by tyn1998**
## Brief Information

This pull request is in the type of ([more info](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#type) about types):

- [ ] build
- [ ] ci
- [x] docs
- [ ] feat
- [x] fix
- [ ] perf
- [x] refactor
- [ ] test

Related issues ([all available keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

- resolve #515

## Details

- [x] refactor DeveloperActiInflTrend & RepoActiInflTrend
- [x] refactor RepoHeaderLabelsView
- [x] refactor other components in repo page
- [ ] ~"Influence" -> "OpenRank" (code object names, docs, ...)~(will handle this in another PR)
- [ ] ~add `attention` and `bus_factor`~(will handle this in another PR)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/hypertrons/hypertrons-crx/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/hypertrons/hypertrons-crx)

## Others

N.A.

### CODE REVIEW

File: src/api/repo.ts [hunk 1]
```diff
@@ -2,6 +2,94 @@ import { HYPERTRONS_OSS_XLAB_ENDPOINT, OSS_XLAB_ENDPOINT } from '../constant';
...
+
+export const getMergedCodeDeletion = async (repo: string) => {
+  return await getMetricByName(repo, 'merged_code_deletion');
+};
+
+export const getMergedCodeSum = async (repo: string) => {
+  return await getMetricByName(repo, 'merged_code_sum');
+};
+
+export const getRepoDetail = async (repo: string) => {
```
Comment (frank-zsy) [inline src/api/repo.ts:88]: Why do you still need repo detail in Hypercrx?

Comment (tyn1998) [reply src/api/repo.ts:88]: because I forgot to remove it 😆

File: src/api/repo.ts [hunk 2]
```diff
@@ -2,6 +2,94 @@ import { HYPERTRONS_OSS_XLAB_ENDPOINT, OSS_XLAB_ENDPOINT } from '../constant';
...
+  const res = await request(
+    `${OSS_XLAB_ENDPOINT}/open_digger/github/${repo}/${metricNameMap.get(
+      metric
+    )}.json`
+  );
+  return res.data;
+};
+
+export const getActivity = async (repo: string) => {
+  return await getMetricByName(repo, 'activity');
```
Comment (frank-zsy) [inline src/api/repo.ts:33]: In Node.js, I think `return await Promise` is equal to `return Promise`, am I right? The behavior maybe the same for outside caller.

Comment (tyn1998) [reply src/api/repo.ts]: You are right, thanks for pointing out that!

**Comment by tyn1998** [comment]
Hi guys, can you fetch this PR branch and do some test to confirm that everything works well?

@wj23027 @andyhuang18 @Zzzzzhuzhiwei @lhbvvvvv

**Comment by wj23027** [comment]
All charts works well~
image

**Comment by xgdyp** [comment]
so quick



At 2022-11-15 20:46:57, "wj23027" ***@***.***> wrote:

All charts works well~

**Comment by andyhuang18** [comment]
everything goes well！
微信截图_20221115215533

**Review by wxharry — APPROVED** [review]

**Comment by wxharry** [comment]
LGTM
what a large refactoring

**Pull request merged**